docs/docs/integrations/providers/notdiamond.mdx

```diff
@@ -0,0 +1,41 @@
+# Not Diamond
+
+[Not Diamond](https://notdiamond.ai) offers an AI-powered model router that automatically determines which LLM is best suited to respond to any query,
+improving LLM output quality by combining multiple LLMs into a **meta-model** that learns when to call each LLM.
+
+Not Diamond supports cost and latency tradeoffs, customized router training, and real-time router personalization. Learn more via [the documentation](https://notdiamond.readme.io/).
+
+## Installation
+
+```shell
+pip install langchain notdiamond
+```
+
+## Configuration
+
+- API keys: You need API keys from [Not Diamond](https://app.notdiamond.ai/keys) and any LLM you want to use.
+
+## Quick Start
+
+Not Diamond offers `NotDiamondRoutedRunnable`, allowing you to include routing within your chain.
+
+```python
+from langchain_community.utilities.notdiamond import NotDiamondRoutedRunnable
+
+from notdiamond import NotDiamond, LLMConfig
+
+nd_client = NotDiamond(
+    llm_configs=[LLMConfig("openai", "gpt-4o"), LLMConfig("openai", "gpt-4o-mini")],
+    api_key=os.getenv("NOTDIAMOND_API_KEY"),
+    default=LLMConfig("openai", "gpt-3.5-turbo"),
+)
+
+nd_runnable = NotDiamondRoutedRunnable(nd_client=nd_client)
+# or: chain = nd_runnable | other_runnable | third_runnable
+
+nd_runnable.invoke("Hello there! Not Diamond sent me to you. Which OpenAI model are you?")
+```
+
+```
+content="Hello! I'm based on OpenAI's GPT-3 model. How can I assist you today?" additional_kwargs=....
+```
```

Comment:
This will blow your mind: that response came from gpt-4o-mini! This output shows the outputted object in a style similar to Langchain's other docs.

Reply:
Also, with that context in mind I understood the response to mean "I'm based on that original architecture" moreso than "you have queried GPT-3." For example, GPT-4 is based on GPT-3 in part, as is 3.5-turbo etc.

Reply:
I guess my point was whether we can use a different query? As this demo example does not seem to show us actually helping produce a high-quality answer. And then instead of trying to shoehorn showing the model selection this way, can we perhaps indicate the specific model that was called in some other way (unless doing so would be awkward in this context)?

Reply:
Ah noted and updating!